plugins/modules/pacemaker_stonith.py

```diff
@@ -0,0 +1,213 @@
+    type: int
+    default: 300
+'''
+
+EXAMPLES = '''
+- name: Create virtual-ip STONITH
+  community.general.pacemaker_stonith:
+    state: present
+    name: virtual-stonith
+    stonith_type: fence_virt
+    stonith_options:
+      - "pcmk_host_list=f1"
+    stonith_operations:
+      - operation_action: monitor
+        operation_options:
+          - "interval=30s"
+'''
+
+RETURN = '''
+cluster_stonith:
+  description: The cluster STONITH output message.
+  type: str
+  sample: ""
+  returned: always
+'''
+
+from ansible_collections.community.general.plugins.module_utils.module_helper import StateModuleHelper
+from ansible_collections.community.general.plugins.module_utils.pacemaker import pacemaker_runner
+
+
+class PacemakerStonith(StateModuleHelper):
+    module = dict(
+        argument_spec=dict(
+            state=dict(type='str', default='present', choices=['present', 'absent', 'enabled', 'disabled']),
+            name=dict(type='str', required=True),
+            stonith_type=dict(type='str'),
+            stonith_options=dict(type='list', elements='str', default=[]),
+            stonith_operations=dict(type='list', elements='dict', default=[], options=dict(
+                operation_action=dict(type='str'),
+                operation_options=dict(type='list', elements='str'),
+            )),
+            stonith_metas=dict(type='list', elements='str'),
+            stonith_argument=dict(type='dict', options=dict(
+                argument_action=dict(type='str', choices=['before', 'after', 'group']),
+                argument_options=dict(type='list', elements='str'),
+            )),
+            agent_validation=dict(type='bool', default=False),
+            wait=dict(type='int', default=300),
+        ),
+        required_if=[('state', 'present', ['stonith_type', 'stonith_options'])],
+        supports_check_mode=True
+    )
+
+    def __init_module__(self):
+        self.runner = pacemaker_runner(self.module)
+        self.vars.set('previous_value', self._get()['out'])
+        self.vars.set('value', self.vars.previous_value, change=True, diff=True)
+
+    def __quit_module__(self):
+        self.vars.set('value', self._get()['out'])
+
+    def _process_command_output(self, fail_on_err, ignore_err_msg=""):
+        def process(rc, out, err):
+            if fail_on_err and rc != 0 and err and ignore_err_msg not in err:
+                self.do_raise('pcs failed with error (rc={0}): {1}'.format(rc, err))
+            out = out.rstrip()
+            return None if out == "" else out
+        return process
+
+    def _get(self):
+        with self.runner('cli_action state name') as ctx:
+            result = ctx.run(cli_action='stonith', state='status')
+            return dict([('rc', result[0]),
+                         ('out', result[1] if result[1] != "" else None),
+                         ('err', result[2])])
+
+    def fmt_stonith_resource(self):
+        return dict([("resource_name", self.vars.stonith_type)])
```

Comment:
Also

```suggestion
        return dict(resource_name=self.vars.stonith_type)
```
Or

```suggestion
        return {"resource_name": self.vars.stonith_type} 
```
Same on the function below